### ISSUE TITLE
Serialize IEnumerable as embeded issue

### ISSUE THREAD

**Issue by xotahal** [SphtKr/JSONAPI.NET#48]
I have problem if class has property `ICollection<SomeClass> foo` and this property has attribute `[SerializeAs(SerializeAsOptions.Embedded)]`. There is  `switch` with `case SerializeAsOptions.Embedded` in `JsonApiFormatter.Serialize(..)`.

That case contains:
`this.Serialize(prop.GetValue(value, null), writeStream, writer, serializer, aggregator);`

But property should be serialized as many.

There could be something like that:
```
var propValue = prop.GetValue(value, null);

if (_modelManager.IsSerializedAsMany(propValue.GetType()))
    this.SerializeMany(propValue, writeStream, writer, serializer, aggregator);
else
    this.Serialize(propValue, writeStream, writer, serializer, aggregator);
```

**Comment by SphtKr** [comment]
Yes, unfortunately `SerializeAsOptions.Embedded` is not fully supported yet. I'm concerned about rushing this as it may introduce complications both for design and implementation (what about recursion? what about *de*serializing?), but we can tag this for 0.4.0 and see if it fits.

**Comment by SphtKr** [comment]
I'm calling this a bug for now, but it may get rolled into a larger enhancement to complete implementation of `SerializeAs.Embedded`.
